tests/xharness/BCLTestImporter/BCLTestProjectGenerator.cs

```diff
@@ -201,508 +134,60 @@ public class BclTestProjectInfo {
-			}
+		public Func<string, Guid> GuidGenerator {
+			get => TemplatedProject.GuidGenerator;
+			set => TemplatedProject.GuidGenerator = value;
 		}
 
-		/// <summary>
-		/// Returns the path to be used to store the project file depending on the platform.
-		/// </summary>
-		/// <param name="projectName">The name of the project being generated.</param>
-		/// <param name="platform">The supported platform by the project.</param>
-		/// <returns></returns>
-		internal string GetProjectPath (string projectName, Platform platform)
-		{
-			switch (platform) {
-			case Platform.iOS:
-				return Path.Combine (OutputDirectoryPath, $"{projectName}.csproj");
-			case Platform.TvOS:
-				return Path.Combine (OutputDirectoryPath, $"{projectName}-tvos.csproj");
-			case Platform.WatchOS:
-				return Path.Combine (OutputDirectoryPath, $"{projectName}-watchos.csproj");
-			case Platform.MacOSFull:
-				return Path.Combine (OutputDirectoryPath, $"{projectName}-mac-full.csproj");
-			case Platform.MacOSModern:
-				return Path.Combine (OutputDirectoryPath, $"{projectName}-mac-modern.csproj");
-			default:
-				return null;
+		public string iOSMonoSDKPath {
+			get {
+				var locator = AssemblyLocator as AssemblyLocator;
+				return locator?.iOSMonoSDKPath;
 			}
-		}
-		
-		internal string GetProjectPath (string projectName, WatchAppType appType)
-		{
-			switch (appType) {
-			case WatchAppType.App:
-				return Path.Combine (OutputDirectoryPath, $"{projectName}-watchos-app.csproj");
-			default:
-				return Path.Combine (OutputDirectoryPath, $"{projectName}-watchos-extension.csproj");
-			}
-		}
-
-		/// <summary>
-		/// Returns the path to be used to store the projects plist file depending on the platform.
-		/// </summary>
-		/// <param name="rootDir">The root dir to use.</param>
-		/// <param name="platform">The platform that is supported by the project.</param>
-		/// <returns></returns>
-		internal static string GetPListPath (string rootDir, Platform platform)
-		{
-			switch (platform) {
-			case Platform.iOS:
-				return Path.Combine (rootDir, "Info.plist");
-			case Platform.TvOS:
-				return Path.Combine (rootDir, "Info-tv.plist");
-			case Platform.WatchOS:
-				return Path.Combine (rootDir, "Info-watchos.plist");
-			case Platform.MacOSFull:
-			case Platform.MacOSModern:
-				return Path.Combine (rootDir, "Info-mac.plist");
-			default:
-				return Path.Combine (rootDir, "Info.plist");
+			set {
+				var locator = AssemblyLocator as AssemblyLocator;
+				if (locator != null)
+					locator.iOSMonoSDKPath = value;
 			}
 		}
 
-		internal static string GetPListPath (string rootDir, WatchAppType appType)
-		{
-			switch (appType) {
-				case WatchAppType.App:
-					return Path.Combine (rootDir, "Info-watchos-app.plist");
-				default:
-					return Path.Combine (rootDir, "Info-watchos-extension.plist");
+		public string MacMonoSDKPath {
+			get {
+				var locator = AssemblyLocator as AssemblyLocator;
+				return locator?.MacMonoSDKPath;
```

Comment:
I think it's safer to make the property of type `AssemblyLocator` and forget about the interface since we are not even injecting it at this point?

Reply:
We have it, but in this case, this class is specific for the xamarin-macios project. I don't want the interface to have the mono sdk paths in the general case when running on helix, does not make much sense